File: lib/data/models/response/product_response.dart
// Hand-written DTO — matches backend ProductResponse JSON exactly

class ProductResponse {
  final int id;
  final String? categoryName;
  final String name;
  final String? description;
  final double? basePrice;
  final List<String> imageUrls;
  final bool isActive;

  const ProductResponse({
    required this.id,
    this.categoryName,
    required this.name,
    this.description,
    this.basePrice,
    required this.imageUrls,
    required this.isActive,
  });

  factory ProductResponse.fromJson(Map<String, dynamic> json) =>
      ProductResponse(
        id: (json['id'] as num).toInt(),
        categoryName: json['categoryName'] as String?,
        name: json['name'] as String? ?? '',
        description: json['description'] as String?,
        basePrice: (json['basePrice'] as num?)?.toDouble(),
        imageUrls: (json['imageUrls'] as List<dynamic>?)
                ?.map((e) => e as String)
                .toList() ??
            [],
        isActive: json['isActive'] as bool? ?? true,
      );
}

File: backend/src/main/java/com/quyenauto/product/dto/ProductResponse.java
package com.quyenauto.product.dto;

import com.quyenauto.product.entity.Product;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
public class ProductResponse {
    private Long id;
    private Long categoryId;
    private String categoryName;
    private String name;
    private String description;
    private String specifications;
    private BigDecimal basePrice;
    private Boolean isActive;
    private List<String> imageUrls;

    public static ProductResponse from(Product p) {
        return ProductResponse.builder()
                .id(p.getId())
                .categoryId(p.getCategory() != null ? p.getCategory().getId() : null)
                .categoryName(p.getCategory() != null ? p.getCategory().getName() : null)
                .name(p.getName())
                .description(p.getDescription())
                .specifications(p.getSpecifications())
                .basePrice(p.getBasePrice())
                .isActive(p.getIsActive())
                .imageUrls(p.getImages().stream().map(img -> img.getImageUrl()).toList())
                .build();
    }
}
